src/routes/safe/components/Settings/SpendingLimit/utils.ts

```diff
@@ -0,0 +1,142 @@
+import GnosisSafeSol from '@gnosis.pm/safe-contracts/build/contracts/GnosisSafe.json'
+import { BigNumber } from 'bignumber.js'
+
+import SpendingLimitModule from 'src/logic/contracts/artifacts/AllowanceModule.json'
+import generateBatchRequests from 'src/logic/contracts/generateBatchRequests'
+import { SENTINEL_ADDRESS } from 'src/logic/contracts/safeContracts'
+import { web3ReadOnly } from 'src/logic/wallets/getWeb3'
+import { SPENDING_LIMIT_MODULE_ADDRESS } from 'src/utils/constants'
+
+export const KEYCODES = {
+  TAB: 9,
+  SHIFT: 16,
+}
+
+export const fromTokenUnit = (amount: string, decimals: string | number): string =>
+  new BigNumber(amount).times(`1e-${decimals}`).toFixed()
+
+export const toTokenUnit = (amount: string, decimals: string | number): string => {
+  const amountBN = new BigNumber(amount).times(`1e${decimals}`)
+  const [, amountDecimalPlaces] = amount.split('.')
+
+  if (amountDecimalPlaces?.length >= +decimals) {
+    return amountBN.toFixed(0, BigNumber.ROUND_DOWN)
+  }
+
+  return amountBN.toFixed()
+}
+
+export const adjustAmountToToken = (amount: string, decimals: string | number): string => {
+  const amountBN = new BigNumber(amount)
+  const [, amountDecimalPlaces] = amount.split('.')
+
+  if (amountDecimalPlaces?.length >= 18) {
+    return amountBN.toFixed(+decimals, BigNumber.ROUND_DOWN)
+  }
+
+  return amountBN.toFixed()
+}
```

Comment:
replace these definitions with what will be implemented in https://github.com/gnosis/safe-react/pull/1444#pullrequestreview-505794722

Reply:
done in [`6deb619` (#1351)](https://github.com/gnosis/safe-react/pull/1351/commits/6deb6190c11d714957cf506990235dc85a656c7b)